pkg/dockerregistry/server/blobdescriptorservice.go

```diff
@@ -25,6 +38,155 @@ type blobDescriptorService struct {
 	distribution.BlobDescriptorService
 }
 
+// Stat returns a a blob descriptor if the given blob is either linked in repository or is referenced in
+// corresponding image stream. This method is invoked from inside of upstream's linkedBlobStore. It expects
+// a proper repository object to be set on given context by upper openshift middleware wrappers.
 func (bs *blobDescriptorService) Stat(ctx context.Context, dgst digest.Digest) (distribution.Descriptor, error) {
-	return dockerRegistry.BlobStatter().Stat(ctx, dgst)
+	repo, repoFound := RepositoryFrom(ctx)
+	if !repoFound {
+		context.GetLogger(ctx).Errorf("failed to retrieve repository from context")
+	}
+
+	// if there is a repo layer link, return its descriptor
+	desc, err := bs.BlobDescriptorService.Stat(ctx, dgst)
```

Comment:
Yes. That's how [the upstream code](https://github.com/miminar/distribution/blob/master/registry/storage/registry.go#L292) currently looks like. We don't have a warranty it will stay this way. What we can do is add a unit tests invoking various stat/get/create methods and ensuring this or some mock blobDescriptorService gets called. Would that be a sufficient guaranty?